### PR TITLE
fix: add compound variant for outline secondary buttons

### PR DESCRIPTION
**Pull request by avirati**
Problem: 
```tsx
<Button theme="secondary" appearance="outline">Button</Button>
```
The above rendered a button with `color: rgb(0,0,0)` when it should have rendered one with `color: $primaryDark`

Reason: 
Compound Variant for this combination was absent. Hence it was falling back to default color

Fix: 
Adds a compound variant to remedy that

Preview :

https://user-images.githubusercontent.com/1936119/160368094-fe8cedb3-9aa3-4416-a491-93dbd9f09048.mov

### CODE REVIEW

File: src/components/button/Button.tsx [hunk 1]
```diff
@@ -178,6 +178,15 @@ export const StyledButton = styled('button', {
...
         opacify(-0.35, 'white')
       )
+    },
+    {
+      theme: 'secondary',
+      appearance: 'outline',
+      css: getButtonOutlineVariant(
+        '$primaryDark',
+        darken(0.05, theme.colors.primaryDark.value),
+        darken(0.1, theme.colors.primaryDark.value)
```
Comment (thomasdigby) [inline src/components/button/Button.tsx:188]: Could we match the values from our secondary solid button? https://github.com/Atom-Learning/components/pull/311/files#diff-dbb0a2706cf9edd17c0ff507bbe8bc37e950d6e7cdb77116e7ca9d5d0c29aa72R135

Comment (avirati) [reply src/components/button/Button.tsx]: Makes sense. Done ✔️

**Comment by kerryambt** [comment]
can we update the mdx? :)

**Comment by avirati** [comment]
> can we update the mdx? :)

Done, thanks

**Comment by kerryambt** [comment]
can you update the appearance section of the mdx to state there is an outline variant for secondary? sorry to be a pain :)

**Comment by avirati** [comment]
> can you update the appearance section of the mdx to state there is an outline variant for secondary? sorry to be a pain :)

Thought of doing that but then if we do this, we would need to have all combinations. That might not look good. Buttons with their individual props have clear demonstration. A reader should be able to expect the desired results

**Comment by kerryambt** [comment]
> > can you update the appearance section of the mdx to state there is an outline variant for secondary? sorry to be a pain :)
> 
> Thought of doing that but then if we do this, we would need to have all combinations. That might not look good. Buttons with their individual props have clear demonstration. A reader should be able to expect the desired results

I just mean here where it says `There are the available outline variations for the primary and neutral themes.` we should add `secondary` to that list, because we don't have all combinations available atm. no need to update the preview :)

image

**Comment by avirati** [comment]
Ah okay. Updated 👍

**Review by kerryambt — APPROVED** [review]

**Review by thomasdigby — APPROVED** [review]

**Pull request merged**